response_readers.go

```diff
@@ -0,0 +1,24 @@
+package opensearchtools
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ReadAggregationResult generically reads a sub bucket from a AggregationResultSet
+// and parses it into the passed aggregation response.
+// subAggResponse can be any pointer type.
+func ReadAggregationResult[A any, P PtrTo[A], R AggregationResultSet](name string, aggResponse R, subAggResponse P) error {
```

Comment:
Maybe this and the `AggregationResultSet` definition should be put in a `aggregation_results.go` file and the `ReadDocument` function can stay in `document.go`. Any thoughts?

Reply:
I like that better than having a separate file with the "reader" functions for all of the various things.